lib/style-property-matcher.js

```diff
@@ -24,41 +24,71 @@ var StylePropertyMatcher = (function (window, document) { // jshint ignore:line
                navigator.userAgent && !navigator.userAgent.match('CriOS') && 
                window.getMatchedCSSRules;
 
+
+    /**
+     * A common use case is using 'content' CSS property, which has <string> type value.
+     * In the result of getComputedStyle, Webkit based browsers encloses such value with single quotes.
+     * Other browsers encloses such value with double quotes.
+     * In order to normalize this, we do a quick check here.
+     */
+    var quote, useFallback = false;
+
+    (function() {
+        var root = document.createElement('div');
+        var styleEl = document.createElement('style');
+        styleEl.appendChild(document.createTextNode('#EXTCSS_TEMP::after{content:""}'));
+        var testEl = document.createElement('div');
+        testEl.setAttribute('style', 'display:none!important;');
+        testEl.id = "EXTCSS_TEMP";
+        root.appendChild(styleEl);
+        root.appendChild(testEl);
+        document.documentElement.appendChild(root);
+        var propValue = window.getComputedStyle(testEl, ':after').getPropertyValue('content');
+        if(propValue.length === 0 && typeof window.getMatchedCSSRules == 'function') {
+            useFallback = true;
+            propValue = window.getMatchedCSSRules(testEl, ':after')[0].style.getPropertyValue('content');
+        }
+        quote = propValue.charAt(0);
+        document.documentElement.removeChild(root);
+    })();
+
+
     /**
      * There is a known issue in Safari browser:
      * getComputedStyle(el, ":before") is empty if element is not visible.
      * 
      * To circumvent this issue we use getMatchedCSSRules instead.
+     *
+     * It appears that getMatchedCSSRules sorts the CSS rules
+     * in increasing order of specifities of corresponding selectors.
      */
-    var getPseudoElementComputedStyle = function(element, pseudoElement) {
-        var styles = [];
-        var cssRules = window.getMatchedCSSRules(element, pseudoElement) || [];
-
-        var iCssRules = cssRules.length;
-        while (iCssRules--) {
-            var style = cssRules[iCssRules].style;
-            var iStyle = style.length;
-            while (iStyle--) {
-                var name = style[iStyle];
-                styles[name] = style.getPropertyValue(name);
-                styles.push(name);
-            }
+    function CSSStyleDec(cssRules) {
```

Comment:
To be sure, here it is not trying to mimick CssStyleDeclaration, It's only providing a method `getPropertyValue` which we use.

`getComputedStylePropertyValue` needs to return an object which has a property `getPropertyValue` having its value a certain function. If we move the function's declaration inside `getComputedStylePropertyValue`, such a function needs to be declared every time `getComputedStylePropertyValue` is called, which I tried to avoid by using prototype.getPropertyValue.